File: src/creational/factory/typescript/classes/factories/BicycleFactory.ts
import { Vehicle } from '../interfaces/VehicleProtocol';
import { Bicycle } from '../Vehicles';
import { VehicleFactoryProtocol } from './VehicleFactory';

export class BicycleFactory extends VehicleFactoryProtocol {
  getVehicle(vehicleName: string): Vehicle {
    return new Bicycle(vehicleName);
  }
}

File: src/creational/factory/python/classes/factories/bicycle_factory.py
from classes.interfaces.vehicle_factory_protocol import VehicleFactoryProtocol
from classes.interfaces.vehicle_protocol import Vehicle
from classes.vehicles import Bicycle


class BicycleFactory(VehicleFactoryProtocol):
    def get_vehicle(self, vehicle_name: str) -> Vehicle:
        return Bicycle(vehicle_name)
